addons/mrp/tests/test_order.py

```diff
@@ -3824,3 +3824,42 @@ def test_mo_duration_expected(self):
         production.button_mark_done()
 
         self.assertEqual(production.workorder_ids.duration_expected, init_duration_expected + 5)
+
+    def test_multi_edit_start_date_wo(self):
+        """
+        Test setting the start date for multiple workorders, checking if the finish date
+        will be set too. As if the finish date is not set the planned workorder will not
+        be shown in planning gantt view
+        """
+        mo = self.env['mrp.production'].create({
+            'product_id': self.product.id,
+            'product_uom_id': self.bom_1.product_uom_id.id,
+        })
+
+        wos = self.env['mrp.workorder'].create([
+            {
+                'name': 'Test order',
+                'workcenter_id': self.workcenter_1.id,
+                'product_uom_id': self.bom_1.product_uom_id.id,
+                'production_id': mo.id,
+                'duration_expected': 0.0
+            },
+            {
+                'name': 'Test order2',
+                'workcenter_id': self.workcenter_2.id,
+                'product_uom_id': self.bom_1.product_uom_id.id,
+                'production_id': mo.id,
+                'duration_expected': 0.0
+            }
+        ])
+
+        dt = datetime.now()
+        wos.date_planned_start = dt
+
+        finish_date = wos[0]._calculate_date_planned_finished(dt)
```

Comment:
In the test, you should replicate the use case so that it behaves in the same way as in the user interface. Therefore, you cannot manually call the "_calculate_date_planned_finished" function.

Reply:
This call is not part of the use case I am just using it as a utility function so I can assert values to make sure that the finish date has been set correctly. `wos.date_planned_finished` which is part of the use case is getting calculated automatically when I set the `date_planned_start`.

Reply:
alternatively, you could just use dt since we already know it should be dt

and maybe you can change the duration_expected and add in the assertEqual of wos[?].date_planned_finished  + timedelta(minutes=number of minutes of the duraction_expected)

Reply:
something like :
```diff
--git a/addons/mrp/tests/test_order.py b/addons/mrp/tests/test_order.py
index 4210c0ade2b6..5b273750a102 100644
--- a/addons/mrp/tests/test_order.py
+++ b/addons/mrp/tests/test_order.py
@@ -3842,24 +3842,22 @@ class TestMrpOrder(TestMrpCommon):
                 'workcenter_id': self.workcenter_1.id,
                 'product_uom_id': self.bom_1.product_uom_id.id,
                 'production_id': mo.id,
-                'duration_expected': 0.0
+                'duration_expected': 1.0
             },
             {
                 'name': 'Test order2',
                 'workcenter_id': self.workcenter_2.id,
                 'product_uom_id': self.bom_1.product_uom_id.id,
                 'production_id': mo.id,
-                'duration_expected': 0.0
+                'duration_expected': 2.0
             }
         ])
 
-        dt = datetime.now()
+        dt = datetime(2022, 6, 15, 12)
         wos.date_planned_start = dt
 
-        finish_date = wos[0]._calculate_date_planned_finished(dt)
-
         self.assertEqual(wos[0].date_planned_start, dt)
         self.assertEqual(wos[1].date_planned_start, dt)
 
-        self.assertEqual(wos[0].date_planned_finished, finish_date)
-        self.assertEqual(wos[1].date_planned_finished, finish_date)
+        self.assertEqual(wos[0].date_planned_finished, dt + timedelta(minutes=1))
+        self.assertEqual(wos[1].date_planned_finished, dt + timedelta(minutes=2))
```

Reply:
Alright got what you mean, my bad I was thinking of something else.